lib/tree_sitter/mixins/language.rb

```diff
@@ -0,0 +1,156 @@
+      def search_for_lib(name)
+        files = [
+          name,
+          "tree-sitter-#{name}",
+          "libtree-sitter-#{name}",
+        ].map { |v| "#{v}.#{ext}" }
+
+        res = lib_dirs
+          .product(files)
+          .find do |dir, so|
+            path = dir / so
+            path = dir / name / so if !path.exist?
+            break path.expand_path if path.exist?
+          end
+        res.is_a?(Array) ? nil : res
+      end
+
+      # Generates a string message on where parser lookup happens.
+      #
+      # @return [String] A pretty message.
+      def search_lib_message
+        indent = 2
+        pretty = ->(arr) {
+          if arr
+            arr
+              .compact
+              .map { |v| "#{' ' * indent}#{v.expand_path}" }
+              .join("\n")
+          end
+        }
+        <<~MSG
+          From ENV['TREE_SITTER_PARSERS']:
+          #{pretty.call(ENV_PARSERS)}
+
+          From TreeStand.config.parser_path:
+          #{pretty.call([TreeStand.config.parser_path])}
+
+          From Defaults:
+          #{pretty.call(LIBDIRS)}
+        MSG
+      end
+
+      # Load a language from configuration or default lookup paths.
+      #
+      # @example Load java from default paths
+      #    # This will look for:
+      #    #
+      #    #   tree-sitter-parsers/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /opt/local/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /opt/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /usr/local/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /usr/lib/(java/)?(libtree-sitter-)?java.{ext}
+      #    #
+      #    java = TreeStand::Parser.language('java')
+      #
+      # @example Load java from a configured path
+      #    # This will look for:
+      #    #
+      #    #   /my/path/(java/)?(libtree-sitter-)?java.{ext}
+      #    #
+      #    TreeStand.config.parser_path = '/my/path'
+      #    java = TreeStand::Parser.language('java')
+      #
+      # @example Load java from environment variables
+      #    # This will look for:
+      #    #
+      #    #   /my/forced/env/path/(java/)?(libtree-sitter-)?java.{ext}
+      #    #   /my/path/(java/)?(libtree-sitter-)?java.{ext}
+      #    #
+      #    # … and the same works for the default paths if `TreeStand.config.parser_path`
+      #    # was `nil`
+      #    ENV['TREE_SITTER_PARSERS'] = '/my/forced/env/path'
+      #    TreeStand.config.parser_path = '/my/path'
+      #    java = TreeStand::Parser.language('java')
+      #
+      # @param name [String] the name of the parser.
+      #   This name is used to load the symbol from the compiled parser, replacing `-` with `_`.
+      # @return [TreeSitter:language] a language object to use in your parsers.
+      # @raise [RuntimeError] if the parser was not found.
+      # @see search_for_lib
+      def language(name)
+        lib = search_for_lib(name)
+
+        if lib.nil?
+          raise <<~MSG
```

Comment:
I don't think you are, but since you often use here-docs, you are passing messages that end with `\n`, and IMHO, they should not.  Use chomp.